### DISCUSSION TITLE
Forelse Without Empty

### DISCUSSION THREAD

**Discussion by ryangurn** [laravel/framework#41477] [Ideas] [answered]
I have been switching most of my code over to using `@forelse` (finally, took a while to realize this existed).

I realized that the `@forelse` directive requires that you use an `@empty` tag to work without erroring. 

There are many times in which `@forelse` would be useful without an `@empty` tag. 

An example of this is that you want to verify that a collection is not empty but if it is empty, there is nothing displayed. 

Using the `@forelse` in this way would reduce the amount of extra blade directive in quite a large way.

The below example shows the current method versus the current method with `@forelse` as well as the functionality I am requesting.

```blade
{{-- current solution with @if & @foreach directives to ensure a collection is non empty before looping --}}
@if(!$collection->isEmpty())
    @foreach($collection as $collect)
        {{-- do something --}}
    @endforeach
@endif

{{-- current solution with @forelse directives to ensure a collection is non empty before looping --}}
@forelse($collection as $collect)
    {{-- do something --}}
@empty
@endforelse

{{-- requested functionality with @forelse directives to ensure a collection is non empty before looping --}}
@forelse($collection as $collect)
    {{-- do something --}}
@endforelse
```

Let me know what you think, I could be misunderstanding the use of the `@forelse` directive but it seems that including an `@empty` directive with nothing to show is a bit useless. 

Its quite possible that there is a better way too!

**Comment by chuckrincon** [comment, marked as answer]
Hi there ✋🏻 .

**TLDR:**
Use `@forelse` if you want an elegant way to handle empty states using Blade.
Use `@foreach` for regular loop statements.

*****

> I could be misunderstanding the use of the @forelse directive

Kind of...

The idea of the `@foresle` is to add an elegant way to handle empty states using Blade.

e.g. This displays a message when the `posts` collections is empty
```blade
@forelse ($posts as $post)
    {{ $post->title }}
@empty
    ¡There are not posts!
@endforelse
```

If you **don't want to display any message** when the when the `posts` collections is empty, you can use the regular `@foreach` blade statement.
e.g. 
```blade
@foreach ($posts as $post)
    {{ $post->title }}
@endforeach
```

**Reply by ryangurn** [reply]
hm, I see. 

The issue that arises is that in most cases when I just use `@forelse` with say and empty collection. I just get errors.

Let me see if I can find an example of when this happens.

**Reply by ryangurn** [reply]
I cannot find an example that works. Thanks so much for your advice! It will greatly help me in making better decisions in the future between `@forelse` and `@foreach`.
